webpack/module-minifier-plugin/src/LocalMinifier.ts

```diff
@@ -56,9 +56,24 @@ export class SynchronousMinifier implements IModuleMinifier {
       return callback(cached);
     }
 
-    const result: IModuleMinificationResult = minifySingleFile(request, this.terserOptions);
-    this._resultCache.set(hash, result);
-
-    callback(result);
+    minifySingleFile(request, this.terserOptions)
+      .then((result: IModuleMinificationResult) => {
+        this._resultCache.set(hash, result);
```

Comment:
Looking at `minifySingleFile` it seems like it's designed already to catch errors and re-output like below. Would we want to cache if there's an error present in the `result`?